src/Enterspeed.Cli/Commands/SourceEntity/DeleteSourceEntityCommand.cs

```diff
@@ -0,0 +1,52 @@
+﻿using Enterspeed.Cli.Api.SourceGroup;
+using Enterspeed.Cli.Services.ConsoleOutput;
+using Enterspeed.Cli.Services.IngestService;
+using MediatR;
+using Microsoft.Extensions.Logging;
+using System.CommandLine.Invocation;
+using System.CommandLine;
+
+namespace Enterspeed.Cli.Commands.SourceEntity;
+
+public class DeleteSourceEntitiesCommand : Command
+{
+    public DeleteSourceEntitiesCommand() : base(name: "delete", "Delete source entity")
+    {
+        AddArgument(new Argument<string>("id", "Id of the source entity to delete") { Arity = ArgumentArity.ExactlyOne });
+        AddOption(new Option<string>(new[] { "--sourceId", "-s" }, "Id of the source") { Arity = ArgumentArity.ExactlyOne });
+    }
+
+    public new class Handler : BaseCommandHandler, ICommandHandler
+    {
+        private readonly IMediator _mediator;
+        private readonly IOutputService _outputService;
+        private readonly IIngestService _ingestService;
+        private readonly ILogger<DeleteSourceEntitiesCommand> _logger;
+
+        public Handler(IMediator mediator, IOutputService outputService, IIngestService ingestService, ILogger<DeleteSourceEntitiesCommand> logger)
+        {
+            _mediator = mediator ?? throw new ArgumentNullException(nameof(mediator));
+            _outputService = outputService;
+            _ingestService = ingestService;
+            _logger = logger;
+        }
+
+        public string SourceId { get; set; }
+        public string Id { get; set; }
+
+        public async Task<int> InvokeAsync(InvocationContext context)
+        {
+            // Find source
```

Comment:
Now that you have a nice comment in the code saying "Find source", it would instead extract these three line to a private method called FindSource